### PR TITLE
Add websocket close functions

### PR DESCRIPTION
**Pull request by dmitry-ee**
I started using this library and noticed that websocket close functions are missing and there's no native and proper way to cleanup `.ws.w` and `.ws.servers` dicts.

Comments appreciated (as i'm not a k-god)

### CODE REVIEW

File: ws-client/ws.q [hunk 1]
```diff
@@ -21,4 +21,12 @@ open0:{[x;y;v]
 
 open:{neg first open0[x;y;.ws.VERBOSE]}                                 //return neg handle for messaging
 
+.ws.close:{[h]
+  if[all(h in key .ws.w;h in key .z.W);hclose h];                       //close handle if h is found both in .ws.w and .z.W (all opened handles)
```
Comment (jonathonmcmurray) [inline ws-client/ws.q:26]: I think we should use `abs h` as within the lib we typically deal with neg handle but I think `hclose` requires positive

```suggestion
  h:abs h;
  if[all(h in key .ws.w;h in key .z.W);hclose h];                       //close handle if h is found both in .ws.w and .z.W (all opened handles)
```

File: ws-client/ws.q [hunk 2]
```diff
@@ -21,4 +21,12 @@ open0:{[x;y;v]
...
+  .ws.w:.ws.w _ h;                                                      //remove h from .ws.w
+  .ws.servers: .ws.servers _ h;                                         //remove h from .ws.servers
```
Comment (jonathonmcmurray) [inline ws-client/ws.q]: think we could just call `.z.wc` here which is already defined to remove from servers/clients tables

```suggestion
  .z.wc h;                                         //remove h from .ws.servers
```

File: ws-client/ws.q [hunk 3]
```diff
@@ -21,4 +21,12 @@ open0:{[x;y;v]
...
+ }
+
+.ws.closea:{ each[.ws.close;(0!.ws.w)[`h]] }                            //close all opened websockets
```
Comment (jonathonmcmurray) [inline ws-client/ws.q]: unusual usage of `each` IMO, this seems more natural to me

```suggestion
.ws.closea:{.ws.close each (0!.ws.w)[`h]}                            //close all opened websockets
```

**Review by jonathonmcmurray — CHANGES_REQUESTED** [review]
thanks for this!

couple of suggestions but looks good

**Comment by dmitry-ee** [comment]
Hey, thanks for not sending me back to the place i came from :)
And thanks for detailed clarifying

File: ws-client/ws.q [hunk 4]
```diff
@@ -21,4 +21,13 @@ open0:{[x;y;v]
...
 open:{neg first open0[x;y;.ws.VERBOSE]}                                 //return neg handle for messaging
 
+.ws.close:{[h]
+  h:abs h;
+  if[all(h in key .ws.w;h in key .z.W);hclose h];                       //close handle if h is found both in .ws.w and .z.W (all opened handles)
+  .ws.w:.ws.w _ h;                                                      //remove h from .ws.w
+  .z.wc h;                                         //remove h from .ws.servers
+ }
+
+.ws.closea:{.ws.close each (0!.ws.w)[`h]}                            //close all opened websockets
```
Comment (jonathonmcmurray) [inline ws-client/ws.q]: Just need to realign these comments and then I'll merge this :+1:

Comment (dmitry-ee) [reply ws-client/ws.q]: Ok, resolved :)

**Review by jonathonmcmurray — APPROVED** [review]

**Pull request merged**